exec/cnex/cell.c

```diff
@@ -239,6 +247,71 @@ BOOL cell_arrayElementExists(const Cell *a, const Cell *e)
     return FALSE;
 }
 
+TString *cell_toString(Cell *c)
+{
+    assert(c->type != cNothing);
+    TString *r = string_newString();
+
+    switch (c->type) {
+        case cAddress:
+            break;
+        case cArray:
+        {
+            size_t x, y;
+            /* ToDo: Fix this to use proper string appends!! */
```

Comment:
You've already got append functions for TString. Use those here instead of an unchecked fixed memory allocation.

Reply:
Removed all C string functions from this case, and use only string_* functions.